src/core/trtserver.h

```diff
@@ -465,32 +466,127 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceManagerNew(
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceManagerDelete(
     TRTSERVER_TraceManager* trace_manager);
 
+/// TRTSERVER_InferenceRequestHeader
```

Comment:
I don't think we need a whole new "class" here after all. Just add methods to Provider. In those methods return an error if it is not a V2 created provider.

Reply:
The idea is that, for benchmarking, the same request header object can be reused for all requests that are consistent. But provider can not be reused because it maintains internal states.

Reply:
Good point, but I don't think "header" should be in the name since that is just an artifact from HTTP/GRPC. How about InferenceRequestOptions to match what we have for server.